File: backend/app/models/agent.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field


class AgentBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    type: str = Field(default='claude-code')
    model: str = Field(default='claude-3-opus')
    timeout: int = Field(default=300, gt=0)
    skills: List[str] = Field(default_factory=list)
    capabilities: List[str] = Field(default_factory=list)


class AgentCreate(AgentBase):
    pass


class AgentUpdate(AgentBase):
    pass


class Agent(AgentBase):
    id: str
    status: str = Field(default='offline')
    created_at: datetime
    updated_at: datetime
    last_seen: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/src/types/agent.ts
export interface AgentType {
  id: string;
  name: string;
  code: string;
  description: string;
  icon: string;
  capabilities: string[];
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface AgentConfig {
  model?: string;
  timeout?: number;
  max_retries?: number;
  skills?: string[];
  [key: string]: unknown;
}

export interface AgentInstance {
  id: string;
  name: string;
  agent_type_id: string;
  status: 'online' | 'offline' | 'busy';
  bridge_url?: string;
  config: AgentConfig;
  last_seen?: string;
  token_usage_today?: number;
  token_usage_month?: number;
  created_at: string;
  updated_at: string;
  // Legacy backward-compat fields (old pages expect these directly)
  type?: string;
  model?: string;
  timeout?: number;
  skills?: string[];
  capabilities?: string[];
}
